### PR TITLE
Move layer type classes to `lifting` package

### LINKED ISSUE #33: Move the layer type classes into another package
For all users of this library that don't define their own transformers, they're likely to never see or use these type classes, so having them in the main package feels pretty unnecessary. @SystemFw suggested they go into their own `lifting` package.

### PR DESCRIPTION
**Pull request by denisrosca**
Closes #33

### CODE REVIEW

**Comment by denisrosca** [comment]
Not sure if the tests in `cats.mtl.laws` and `cats.mtl.laws.discipline` should be moved similar to the type classes.

File: core/src/main/scala/cats/mtl/lifting/FunctorLayer.scala [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package cats
 package mtl
+package lifting
+
+import cats.{Functor, ~>}
```
Comment (LukaJCB) [inline core/src/main/scala/cats/mtl/lifting/FunctorLayer.scala]: Aren't these imports redundant since we're already in the `cats` package?

**Comment by denisrosca** [comment]
@LukaJCB Yep, you are correct. Fixed.

**Comment by LukaJCB** [comment]
Looks good to me, thanks very much! :)

**Comment by LukaJCB** [comment]
@andyscott @kailuowang @SystemFw 
We need to think about versioning soon, I think most agreed that this change is definitely worth it, but the question is what's our versioning scheme going forward.

**Comment by kailuowang** [comment]
just realized that somehow I wasn't watching this repo. I vote for just bump the minor for this change since we are still pre 1.0

**Comment by LukaJCB** [comment]
That is my vote as well, should we merge this PR and move the discussion to its own issue (sorry @denisrosca)

**Comment by SystemFw** [comment]
Is this ready to merge then?

**Pull request merged**